janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchIndex.java

```diff
@@ -181,6 +181,11 @@
      */
     public static final double DEFAULT_GEO_DIST_ERROR_PCT = 0.025;
 
+    /*
+     * A special character to use instead of whitespace, which is not allowed in field names. See #777.
+     */
+    private static char REPLACEMENT_CHAR = '\u2022';
```

Comment:
This is replicated 3 ways across the 3 index backends; any way this can be kept in a single location?

Reply:
I'm happy to do so, but I wasn't sure where a good place was. The most relevant place seems like the IndexProvider interface alongside the mapKey2Field method, but I wasn't sure if others considered it clean to add a constant to an interface. Not to mention it's kind of an implementation detail of each index provider and it doesn't *have* to be the same...but if IndexProvider or another location works, I'll move it.

Reply:
I don't have a strong opinion about this, and you make a good point.

Maybe @sjudeng and/or @davidclement90 have some thoughts on this?

Reply:
Like others static constant, I may put it in org.janusgraph.graphdb.configuration.GraphDatabaseConfiguration ?

Reply:
It could go there but that's mostly public constants and configuration, and this definitely isn't something anyone other than an implementer of IndexProvider should ever need to know about. I'd prefer IndexProvider over GraphDatabaseConfiguration.

Reply:
I decided to put it in IndexProvider, but am open to moving it elsewhere.